packages/kbn-es-query/src/filters/build_filters/types.ts

```diff
@@ -79,7 +74,9 @@ export type Filter = {
     store: FilterStateStore;
   };
   meta: FilterMeta;
-  query?: any; // TODO: can we use the Query type her?
+
+  // TODO: research me! This is being extracted into the top level by translateToQuery. Maybe we can simplify.
+  query?: Record<string, any>;
```

Comment:
this should be DslQuery right ?

Reply:
Theoretically yes, but practically there were some differences.
I think we should actually get rid of this prop altogether, it's used in a very bad way - I'll explain on a call.